admin/spec/components/previews/solidus_admin/ui/table/component_preview.rb

```diff
@@ -17,11 +17,17 @@ def simple
       model_class.new(id: n, name: "Product #{n}", price: n * 10.0, available_on: n.days.ago)
     end
 
+    page = Struct.new(:records, :number, :next_param, :first?, :last?).new(rows, 1, '#', true, false)
+
+    page.records.define_singleton_method(:model) do
+      model_class
+    end
```

Comment:
If there were a way to create a collection proxy from the above-generated array, we wouldn't need to do that. I didn't find a way, but leaving here the comment in case someone has a better idea :slightly_smiling_face:

Reply:
Yeah, I don't like it either. I've also tried to come up with some ideas to simplify this part, but unfortunately, I couldn't find a clearer solution. 😕